Pinta.Core/Effects/HsvColor.cs

```diff
@@ -6,141 +6,66 @@
-		=> lhs.Hue == rhs.Hue && lhs.Saturation == rhs.Saturation && lhs.Value == rhs.Value;
-
-	public static bool operator != (HsvColor lhs, HsvColor rhs)
-		=> !(lhs == rhs);
-
-	public override readonly bool Equals (object? obj)
-		=> obj is HsvColor hsv && this == hsv;
+	public static bool operator == (HsvColor c1, HsvColor c2)
+	{
+		if (Math.Abs (c1.Hue - c2.Hue) < 0.1 &&
+		    Math.Abs (c1.Sat - c2.Sat) < 0.001 &&
+		    Math.Abs (c1.Val - c2.Val) < 0.001)
+			return true;
+		return false;
+	}
 
-	public override readonly int GetHashCode ()
-		=> (Hue + (Saturation << 8) + (Value << 16)).GetHashCode ();
+	public static bool operator != (HsvColor c1, HsvColor c2) => !(c1 == c2);
 
-	public HsvColor (int hue, int saturation, int value)
+	public HsvColor (double hue, double sat, double val)
 	{
 		if (hue < 0 || hue > 360)
 			throw new ArgumentOutOfRangeException (nameof (hue), "must be in the range [0, 360]");
 
-		if (saturation < 0 || saturation > 100)
-			throw new ArgumentOutOfRangeException (nameof (saturation), "must be in the range [0, 100]");
+		if (sat < 0 || sat > 1)
+			throw new ArgumentOutOfRangeException (nameof (sat), "must be in the range [0, 1]");
 
-		if (value < 0 || value > 100)
-			throw new ArgumentOutOfRangeException (nameof (value), "must be in the range [0, 100]");
+		if (val < 0 || val > 1)
+			throw new ArgumentOutOfRangeException (nameof (val), "must be in the range [0, 1]");
 
 		Hue = hue;
-		Saturation = saturation;
-		Value = value;
+		Sat = sat;
+		Val = val;
 	}
 
-	//        public static HsvColor FromColor(Color color)
-	//        {
-	//            RgbColor rgb = new RgbColor(color.R, color.G, color.B);
-	//            return rgb.ToHsv();
-	//        }
-	//
-	//        public Color ToColor()
-	//        {
-	//            RgbColor rgb = ToRgb();
-	//            return Color.FromArgb(rgb.Red, rgb.Green, rgb.Blue);
-	//        }
-
-	public readonly RgbColor ToRgb ()
-	{
-		// HsvColor contains values scaled as in the color wheel.
-		// Scale Hue to be between 0 and 360. Saturation
-		// and value scale to be between 0 and 1.
-		double h = (double) Hue % 360;
-		double s = (double) Saturation / 100;
-		double v = (double) Value / 100;
-
-		double r = 0;
-		double g = 0;
-		double b = 0;
-
-		if (s == 0) {
-			// If s is 0, all colors are the same.
-			// This is some flavor of gray.
-			r = v;
-			g = v;
-			b = v;
-		} else {
-			// The color wheel consists of 6 sectors.
-			// Figure out which sector you're in.
-			double sectorPos = h / 60;
-			int sectorNumber = (int) (Math.Floor (sectorPos));
+	public static HsvColor FromColor (Color c) => c.ToHsv ();
+	public Color ToColor (double alpha = 1) => Color.FromHsv (this);
 
-			// get the fractional part of the sector.
-			// That is, how many degrees into the sector
-			// are you?
-			double fractionalSector = sectorPos - sectorNumber;
+	public static HsvColor FromRgbD (double r, double g, double b) => new Color (r, g, b).ToHsv ();
```

Comment:
These FromRgb() variants seem to be unused, so I'd lean towards just removing them unless you want them for some upcoming changes? I think most code is typically using the various color type rather than passing around separate r/g/b values.